vispy/scene/cameras/cameras.py

```diff
@@ -1128,6 +1009,242 @@ def _update_projection_transform(self, fx, fy):
+        elev = float(elev)
+        self._elevation = min(90, max(-90, elev))
+        self.view_changed()
+
+    @property
+    def azimuth(self):
+        """ The angle of the camera in degrees around the y axis. An angle of
+        0 places the camera within the (y, z) plane.
+        """
+        return self._azimuth
+
+    @azimuth.setter
+    def azimuth(self, azim):
+        azim = float(azim)
+        while azim < -180:
+            azim += 360
+        while azim > 180:
+            azim -= 360
+        self._azimuth = azim
+        self.view_changed()
+
+    @property
+    def roll(self):
+        """ The angle of the camera in degrees around the z axis. An angle of
+        0 places puts the camera upright.
+        """
+        return self._roll
+
+    @roll.setter
+    def roll(self, roll):
+        roll = float(roll)
+        while roll < -180:
+            roll += 360
+        while roll > 180:
+            roll -= 360
+        self._roll = roll
+        self.view_changed()
+
+    def orbit(self, azim, elev):
+        """ Orbits the camera around the center position.
+
+        Parameters
+        ----------
+        azim : float
+            Angle in degrees to rotate horizontally around the center point.
+        elev : float
+            Angle in degrees to rotate vertically around the center point.
+        """
+        self.azimuth += azim
+        self.elevation = np.clip(self.elevation + elev, -90, 90)
+        self.view_changed()
+
+    def _update_rotation(self, event):
+        """Update rotation parmeters based on mouse movement"""
+        p1 = event.mouse_event.press_event.pos
+        p2 = event.mouse_event.pos
+        if self._event_value is None:
+            self._event_value = self.azimuth, self.elevation
+        self.azimuth = self._event_value[0] - (p2 - p1)[0] * 0.5
+        self.elevation = self._event_value[1] + (p2 - p1)[1] * 0.5
+
+    def _rotate_tr(self):
+        """Rotate the transformation matrix based on camera parameters"""
+        up, forward, right = self._get_dim_vectors()
+        self.transform.rotate(self.elevation, -right)
+        self.transform.rotate(self.azimuth, up)
+
+    def _dist_to_trans(self, dist):
+        """Convert mouse x, y movement into x, y, z translations"""
+        sro, saz, sel = list(map(sind, (self.roll, self.azimuth,
+                                        self.elevation)))
+        cro, caz, cel = list(map(cosd, (self.roll, self.azimuth,
+                                        self.elevation)))
+        dx = (+ dist[0] * (cro * caz + sro * sel * saz)
+              + dist[1] * (sro * caz - cro * sel * saz))
+        dy = (+ dist[0] * (cro * saz - sro * sel * caz)
+              + dist[1] * (sro * saz + cro * sel * caz))
+        dz = (- dist[0] * sro * cel + dist[1] * cro * cel)
+        return dx, dy, dz
+
+
+class ArcballCamera(Base3DRotationCamera):
+    """ 3D camera class that orbits around a center point while
+    maintaining a view on a center point.
```

Comment:
With the turntable there is a notion of up, and by interaction the azimuth and elevantion are modified. With an arcball camera you go all over the place :)